hoodie-common/src/main/java/com/uber/hoodie/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -99,10 +107,12 @@ public Schema getSchema() {
         output.writeInt(size);
         // Write the content
         output.write(temp.toByteArray());
+        // TODO : remove entries from this list to reduce memory pressure
+        //itr.remove();
```

Comment:
It would be ideal to remove the records that have been converted to bytes here but this requires a refactor of the way tests are written. Also, to make sure these records are never used again.

Reply:
how many tests is this.. Can't we make a copy of the list from the test, before we pass it in here?

Reply:
Many tests, let me check on what can be changed in the tests, waited to get your thoughts on this and looks like you're on the same page of removing these entries once used.